firebase-database/src/main/java/com/google/firebase/database/core/ServerValues.java

```diff
@@ -145,22 +143,22 @@ public void visitChild(ChildKey name, Node child) {
             }
           });
       if (!holder.getRootNode().getPriority().equals(priority)) {
-        return holder.getRootNode().updatePriority(priority);
+        return holder.getRootNode().updatePriority(PriorityUtilities.parsePriority(priority));
       } else {
         return holder.getRootNode();
       }
     }
   }
 
   public static CompoundWrite resolveDeferredValueMerge(
-      CompoundWrite merge, Node existing, final Map<String, Object> serverValues) {
+      CompoundWrite merge, SyncTree syncTree, Path path, final Map<String, Object> serverValues) {
     CompoundWrite write = CompoundWrite.emptyWrite();
```

Comment:
Interesting. In iOS this starts as merge and is modified with addWrite only where the resolved value changes. Is this a pessimisation?

Reply:
I tried to make it match iOS just now but it broke a bunch of tests.